src/lib/SmartTokens/Contracts/ProtocolParams.hs

```diff
@@ -40,30 +26,30 @@ mkProtocolParametersMinting = plam $ \oref ctx -> P.do
   ctxF <- pletFields @'["txInfo", "scriptInfo"] ctx
   infoF <- pletFields @'["inputs", "mint"] ctxF.txInfo
   scriptInfoF <- pletFieldsMinting ctxF.scriptInfo
-  let ownCS = scriptInfoF._0 
+  let ownCS = scriptInfoF._0
   mintedValue <- plet $ pfromData infoF.mint
   let ownTkPairs = ptryLookupValue # ownCS # mintedValue
-  -- Enforce that only a single token name is minted for this policy 
+  -- Enforce that only a single token name is minted for this policy
   ownTkPair <- plet (pheadSingleton # ownTkPairs)
   ownTokenName <- plet (pfstBuiltin # ownTkPair)
   ownNumMinted <- plet (psndBuiltin # ownTkPair)
   pvalidateConditions
-    [ ownTokenName #== pprotocolParamsTokenData
-    , ownNumMinted #== pconstantData 1
-    , phasUTxO # oref # infoF.inputs
+    [ ptraceInfoIfFalse "token name mismatch" $ ownTokenName #== pprotocolParamsTokenData
+    , ptraceInfoIfFalse "incorrect minted amount" $ ownNumMinted #== pconstantData 1
+    , ptraceInfoIfFalse "init UTxO not present" $ phasUTxO # oref # pfromData infoF.inputs
```

Comment:
Bug 2. This list was data encoded so we should have applied fromData to pasList to convert from BuiltinData to BuiltinList